### ISSUE TITLE
Multiplayer remote player kicked when they "see" my machete.

### ISSUE THREAD

**Issue by DrLucky007** [SilentChaos512/Silent-Gear#143]
## Versions

- **Silent Gear**: SilentGear-1.15.2-1.7.2+96
- **Silent's Gems**: SilentGems-1.15.2-3.5.5+88
- **Silent Lib**: SilentLib-1.15.2-4.6.1+49
- **Forge**: 31.2.9
- **Modpack**: MC-Techcolonies 3.4
- **Optifine Installed**: No

## Expected Behavior

The player to not get kicked.

## Actual Behavior

Remote player gets kicked with one of a variety of messages.  (Null Pointer Exception or Timeout are common).

## Links/Images

Link to my (host) log: https://pastebin.com/WuQnC13S
showing a typical attempt to login by a remote player.

Link to remote player log.  Their system clock seems to be within about 2s of mine: https://pastebin.com/5uaPQZCW

## Steps to Reproduce the Problem

1. Put my machete in my hotbar (Pyrope, Pyrope, Hardened Stone, Iron Rod, Lapis Coated)
2. Walk up to player
3. Select the machete 
4. Player immediately gets kicked from the game
5. Player tries to log back in
6. Player gets kicked
7. Select another tool
8. Player tries to log in
9. Player succeeds.

I believe I've experienced this as a remote player, but I had made a multi-material pick in that case, and it was in my own inventory.  I was subsequently unable to log back in successfully.  I will be asking the host to delete my player files to see if we can replicate that way.

~~It's very likely tied to multi-material tools, since we are using single-material tools frequently without issue.~~ Further testing shows it's very likely tied to the Lapis coating.

It does not appear to affect me as host.

**Comment by DrLucky007** [comment]
Follow up:  I noticed I had a ruby-diamond-ruby pickaxe I used regularly, so I added a lapis coating to an old pick and tested both.

It seems to be the lapis coating.  Remote player was unaffected by the Ruby/Diamond pick, and the Iron/Refined Iron/Refined Iron/Lapis Coated pick instantly kicked him.

Updating original report.

**Comment by macrosblackd** [comment]
Happens with bone tool rods too.

**Comment by SilentChaos512** [comment]
So it might be linked to the Lucky trait? No idea what's happening here.

**Comment by DrLucky007** [comment]
I'll test some theories with a friend tonight.  If it's tied to "Lucky" we can test with an amethyst tool. 
If it's tied to number of materials, we can test that easily.
Also, the modpack updated today, so I'm now testing on:
SilentGear-1.15.2-1.7.5+106
SilentGems-1.15.2-3.5.5+88
SilentLib-1.15.2-4.6.3+55
Forge 31.2.20
MC-Techcolonies 3.5

**Comment by DrLucky007** [comment]
@SilentChaos512  

So we just conducted a test - the host made an amethyst shovel (Shovel Blueprint + Amethyst + Stick).  It kicked me.
He then put a redstone tip on his pick, and that had no effect.

So you're right - it seems to be the "lucky" attribute.
This matches my experience with my machete which had a lucky property kicking players when I was hosting.

Long shot - is it possible that it has something to do with JEI repair recipe integration?

As I mentioned above, I'm now on SilentGear-1.15.2-1.7.5+106 and seeing the same effect.

**Comment by McKitten** [comment]
Just as a note for the strange :

SilentGear-1.15.2-1.7.1+94 is in Vanhelsia 2 modpack and does not seem affected by this. 

Perhaps it is a Gem's interaction triggering it as the pack does not contain it ? Or it could be just one of the many things Vanhelsia team broke with it.

--

**Comment by macrosblackd** [comment]
I'm encountering this IN Valhelsia 2 modpack. So it isn't just a gems issue. The crash doesn't happen when in singleplayer (or as equipping it as host in a lan game). Only connected clients crash when the tool is equipped.

**Comment by SilentChaos512** [comment]
Might be fixed by the change to `AttributeTrait` in e7000a775ce37680870c501ea537ad0a50412b81. If someone could retest with Silent Gear 1.8.0, that would be appreciated.

**Comment by DrLucky007** [comment]
Okay, my friend and I tested this out.  MC-Techcolonies, as above, only change is that we each downloaded and installed [SilentGear-1.15.2-1.8.0+107].

Testing in a new world, we were able to use and handle both an amethyst pick (with inherent luck) and a lapis-coated yellow diamond pick (luck from coating) withotu issue to the remote player.

We're going to continue with our regular world using this version to see if anything crops up, but it looks good!
